### PR TITLE
A rude implemention of settings for Featured Content Items

### PR DESCRIPTION
**Pull request by redundans**
Ok, this is a proof of concept for a very rude re-implementation of settings. The settings are now attributes to a featured area. And as all other featured area stettings it is translated through element data attributes to the customizer script.

The customizer.js script handle all the setting functionality. It renders it, loads it and saves it. That way we do very little harm to the FCM core functions. Maybe we should also refactor some 'featured area to wp.template' logics.

To test this you have to set up settings for each featured area like this:

```php
add_action(
	'after_setup_theme',
	function() {
		add_theme_support(
			'featured-content-manager',
			[
				'featured_areas' => [
					'frontpage' => [
						'title' => 'Förstasidan',
						'settings' => [
							'style' => [
								'type'         => 'select',
								'display_name' => 'Utseende',
								'values'       => [
									'standard' => 'Standard',
									'large'    => 'Stor',
								],
							],
						],
					],
					'secondpage' => [
						'title' => 'Andrasidan',
					]
				],
			]
		);
	},
	11
);
```

For the moment the JS only handles settings of type `select`.

Also the custmizer setting object already has an attribute name `settings` so thats why I use `fsettings` in the plugin.

### CODE REVIEW

File: includes/class-featured-area-control.php [hunk 1]
```diff
@@ -40,6 +40,13 @@ class Featured_Area_Control extends \WP_Customize_Control {
 	 */
 	public $levels = 1;
 
+	/**
+	 * The number of levels to stack featured items.
+	 *
+	 * @var int $levels The levels.
+	 */
+	public $ssettings = [];
```
Comment (lakrisgubben) [inline includes/class-featured-area-control.php]: should be **f**settings?

File: includes/class-featured-content.php [hunk 2]
```diff
@@ -30,6 +30,26 @@ public static function get_featured_areas() {
...
+					$fields[ $field_name ]['name']         = $field['name'];
+					$fields[ $field_name ]['display_name'] = $field['display_name'];
+					$fields[ $field_name ]['type']         = 'select';
+					$fields[ $field_name ]['values']       = $field['values'];
+				}
+			}
+		}
+		return $fields;
+	}
+
```
Comment (lakrisgubben) [inline includes/class-featured-content.php]: can't see this being used anywhere?

**Comment by lakrisgubben** [comment]
Most of the things seem to work great! This is what's left as far as I can tell:

- [ ] make sure the customizer updates when a select is changed
- [ ] Some styling for the selectboxes
- [ ] esc_html on post/term title in rest response + find out what's breaking the json when inserting a post with quotes in the title

**Comment by redundans** [comment]
> Most of the things seem to work great! This is what's left as far as I can tell:
> 
>     * [ ]  make sure the customizer updates when a select is changed
> 
>     * [ ]  Some styling for the selectboxes
> 
>     * [ ]  esc_html on post/term title in rest response + find out what's breaking the json when inserting a post with quotes in the title

I think I got all those issues in my latest 2 pull requests.

**Review by lakrisgubben — APPROVED** [review]
Let's do it!

**Pull request merged**